client/src/lib/dependency-injection/tests/MiddlewareRegistry-test.js

```diff
@@ -129,12 +77,23 @@ describe('MiddlwareRegistry', () => {
 
       registry.sort();
 
-      const factory = registry.getFactory();
+      const matches = registry.getMatchesForContext().map(m => m.factory);
+      const factory = compose(...matches)(service);
```

Comment:
Yes, much cleaner, and a more accurate test to boot. Have updated all the tests and eliminated the `compose` dependency.